sentinel-transport/sentinel-transport-simple-http/src/main/java/com/alibaba/csp/sentinel/transport/heartbeat/client/SimpleHttpClient.java

```diff
@@ -77,57 +90,73 @@ public SimpleHttpResponse post(SimpleHttpRequest request) throws IOException {
         if (request == null) {
             return null;
         }
-        return request(request.getSocketAddress(),
+        return request(request.getProtocol(), request.getSocketAddress(),
             RequestMethod.POST, request.getRequestPath(),
             request.getParams(), request.getCharset(),
             request.getSoTimeout());
     }
 
+	private SimpleHttpResponse request(String protocol, InetSocketAddress socketAddress,
+									   RequestMethod type, String requestPath,
+									   Map<String, String> paramsMap, Charset charset, int soTimeout)
+			throws IOException {
+		Socket socket = null;
+		BufferedWriter writer;
+		try {
+			if (StringUtil.equals(protocol, "http")) {
+				socket = new Socket();
+			} else if (StringUtil.equals(protocol, "https")){
+				sslSocketFactory = getSSLSocketFactory();
+				if (sslSocketFactory != null) {
+					socket = sslSocketFactory.createSocket();
+				} else {
+					socket = new Socket();
```

Comment:
Maybe it doesn't make any sense here?

Reply:
> Maybe it doesn't make any sense here?

Yes, actually, it doesn't work here. It's just used to prevent NPE, but it won't happen in theory. I'll delete it.